WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift

```diff
@@ -0,0 +1,110 @@
+                                                             onCardReaderMessage: { (event) in
+                                                                switch event {
+                                                                case .displayMessage (let message):
+                                                                    onPresentMessage(message)
+                                                                case .waitingForInput (let message):
+                                                                    onPresentMessage(message)
+                                                                case .cardRemoved:
+                                                                    onClearMessage()
+                                                                default:
+                                                                    break
+                                                                }
+                                                             }, onCompletion: { [weak self] result in
+                                                                self?.completePayentIntentCapture(order: order,
+                                                                                                 captureResult: result,
+                                                                                                 onCompletion: onCompletion)
+        })
+
+        ServiceLocator.stores.dispatch(action)
+    }
+
+    func completePayentIntentCapture(order: Order,
+                                    captureResult: Result<PaymentIntent, Error>,
+                                    onCompletion: @escaping (Result<CardPresentReceiptParameters, Error>) -> Void) {
+        switch captureResult {
+        case .failure:
+            let error = CardReaderServiceError.paymentCapture()
+            onCompletion(.failure(error))
+        case .success(let paymentIntent):
+            submitPaymentIntent(siteID: order.siteID,
+                                orderID: order.orderID,
+                                paymentIntent: paymentIntent,
+                                onCompletion: onCompletion)
+        }
+    }
+
+    func submitPaymentIntent(siteID: Int64,
+                             orderID: Int64,
+                             paymentIntent: PaymentIntent,
+                             onCompletion: @escaping (Result<CardPresentReceiptParameters, Error>) -> Void) {
+        let action = WCPayAction.captureOrderPayment(siteID: siteID,
+                                                     orderID: orderID,
+                                                     paymentIntentID: paymentIntent.id) { result in
+
+            guard let receiptParameters = paymentIntent.receiptParameters() else {
+                let error = CardReaderServiceError.paymentCapture()
+
+                DDLogError("⛔️ Payment completed without valid regulatory metadata: \(error)")
```

Comment:
It isn't regulatory in the sense of legal matters, but more a requirement of the card brands. Maybe regulatory -> required